tg/utils.py

```diff
@@ -180,9 +181,24 @@ def notify(
     subprocess.Popen(notify_cmd, shell=True)
 
 
-def truncate_to_len(s: str, target_len: int) -> str:
-    target_len -= sum(map(bool, map(emoji_pattern.findall, s[:target_len])))
-    return s[: max(1, target_len - 1)]
+def string_len_dwc(string: str) -> int:
+    """Returns string len including count for double width characters"""
+    return sum(1 + (unicodedata.east_asian_width(c) in "WF") for c in string)
+
+
+def truncate_to_len(string: str, width: int) -> str:
+    real_len = string_len_dwc(string)
+    if real_len <= width:
+        return string
```

Comment:
this lines look redundant, do we really need them?

Reply:
Doesn't look redundant for me

Reply:
Suggest better alternative in other patch I would appreciate it.